src/main/java/com/alex/eshop/eshop/Service/ItemServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.alex.eshop.eshop.Service;
+
+import com.alex.eshop.eshop.Entity.Item;
+import com.alex.eshop.eshop.ExceptionHandling.NoSuchItemException;
+import com.alex.eshop.eshop.Repository.ItemRepository;
+
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import java.util.*;
+
+@Service
+@Transactional
+public class ItemServiceImpl implements ItemService {
+    private final ItemRepository itemRepository;
+
+    public ItemServiceImpl(ItemRepository itemRepository) {
+        this.itemRepository = itemRepository;
+    }
+
+    @Override
+    public Item getItemWithCategoryInfo(Long id) {
```

Comment:
Function `public Item getItemWithCategoryInfo(Long id);` is good, but it can be improved. **Code can be reduced to 1-2 lines** if use orElseThrow() for Optional.

```
return itemRepository.findById(id)
        .orElseThrow(() -> new NoSuchItemException("There is no item with id " + id));
```